packages/cli/src/util/dev/parse-query-string.ts

```diff
@@ -0,0 +1,41 @@
+export function parseQueryString(querystring?: string) {
+  let map = new Map<string, string[]>();
+  if (!querystring || !querystring.startsWith('?') || querystring === '?') {
+    return map;
+  }
+  const params = querystring.slice(1).split('&');
+  for (let param of params) {
+    const [key, value] = param.split('=');
+
+    let existing = map.get(key);
+    if (!existing) {
+      existing = [];
+      map.set(key, existing);
+    }
+
+    existing.push(value);
```

Comment:
Do you need to `decodeURIComponent()` the value?

Reply:
Ya I believe we need to decode both the key and value.

Reply:
Just to be clear, are suggesting decoding during parse() and encoding during format()?

Reply:
I only tested the parse, but you should probably also `encodeURIComponent()` during format.

Reply:
I updated in [a5aee22](https://github.com/vercel/vercel/pull/8183/commits/a5aee2278269a19d2974e3888330bb720ff3ee01) so this behavior matches `url.parse()` and `url.format()` with decode and encode respectively. Nice catch! 👍